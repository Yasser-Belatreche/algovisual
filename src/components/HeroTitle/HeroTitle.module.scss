.title {
  @apply text-blackShade leading-none font-bold;
  font-size: 65px;
  letter-spacing: -1.5px;

  &.white {
    @apply text-white;
  }
}
