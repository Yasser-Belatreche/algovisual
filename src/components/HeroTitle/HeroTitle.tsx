import React from "react";
import classNames from "classnames/bind";

// styles
import styles from "./HeroTitle.module.scss";

const cx = classNames.bind(styles);

interface Props extends React.HTMLAttributes<HTMLHeadingElement> {
  color?: "black" | "white";
}

const HeroTitle: React.FC<Props> = ({
  children,
  color,
  className: classFromProps,
  ...props
}) => {
  const className = cx("title", { white: color == "white" });

  return (
    <h1 className={`${className} ${classFromProps}`} {...props}>
      {children}
    </h1>
  );
};

export { HeroTitle };
