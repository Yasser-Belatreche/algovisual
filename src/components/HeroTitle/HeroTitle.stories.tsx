import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { HeroTitle } from "./HeroTitle";

export default {
  title: "Global/HeroTitle",
  component: HeroTitle,
  argTypes: {},
} as ComponentMeta<typeof HeroTitle>;

const Template: ComponentStory<typeof HeroTitle> = (args) => (
  <HeroTitle {...args}>Algorithms</HeroTitle>
);

export const Black = Template.bind({});
Black.args = {};

export const White = Template.bind({});
White.args = {
  color: "white",
};
