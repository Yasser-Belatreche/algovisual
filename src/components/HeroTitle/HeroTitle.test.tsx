import { render } from "@testing-library/react";

import { HeroTitle } from "./HeroTitle";

describe("Button compoenent", () => {
  it("should have the black shade color by default", () => {
    const { container } = render(<HeroTitle>My title</HeroTitle>);

    expect(container).toMatchSnapshot();
  });

  it("should have the white color when changing the color prop to white", () => {
    const { container } = render(<HeroTitle color="white">My title</HeroTitle>);

    expect(container).toMatchSnapshot();
  });

  it("should add the class passed to props to the class list", () => {
    const instance = render(
      <HeroTitle className="myCustomClass">My title</HeroTitle>
    );

    const header = instance.getByRole("heading");

    expect(header.classList.length).not.toEqual(1);
    expect(header.classList).toContain("myCustomClass");
    expect(instance.container).toMatchSnapshot();
  });
});
