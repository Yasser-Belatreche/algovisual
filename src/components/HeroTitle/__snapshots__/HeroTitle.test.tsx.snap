// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button compoenent should add the class passed to props to the class list 1`] = `
<div>
  <h1
    class="title myCustomClass"
  >
    My title
  </h1>
</div>
`;

exports[`Button compoenent should have the black shade color by default 1`] = `
<div>
  <h1
    class="title undefined"
  >
    My title
  </h1>
</div>
`;

exports[`Button compoenent should have the white color when changing the color prop to white 1`] = `
<div>
  <h1
    class="title white undefined"
  >
    My title
  </h1>
</div>
`;
